examples/all-clusters-app/all-clusters-common/include/oven-operational-state-delegate.h

```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace OvenCavityOperationalState {
+
+// This is an application level delegate to handle operational state commands according to the specific business logic.
+class OvenCavityOperationalStateDelegate : public OperationalState::Delegate
+{
+private:
+    inline static const Clusters::OperationalState::GenericOperationalState mOperationalStateList[] = {
```

Comment:
Now that this is `inline static`:

Just want to confirm one thing, if you have two clusters on different endpoints (I know you don't allow this today because you check that it is on endpoint == 1, but lets say someone removes that that restriction in the future) will it be a problem that the `OperationsStateList` is static and shared?

I think this is not a problem, but would just like confirmation

Reply:
Yes, I considered this drawback and agree that for this implementation it doesn't matter. I think it is likely ok too with the static list for almost all implementations since this is a list of all possible states and in the case of a double oven (which would naturally have more than 1 endpoint including this cluster) the two ovens would likely share the same set of possible states.